src/Server.php

```diff
@@ -137,6 +137,33 @@ public function handleRequest(ConnectionInterface $conn, Request $request)
             if (strtolower(end($transferEncodingHeader)) === 'chunked') {
                 $stream = new ChunkedDecoder($conn);
             }
+
+            if ($request->hasHeader('Content-Length')) {
+                // remove 'Content-Length' header accordig to: https://tools.ietf.org/html/rfc7230#section-3.3.3
+                $headers = $request->getHeaders();
+                unset($headers['Content-Length']);
+
+                $request = new Request(
+                    $request->getMethod(),
+                    $request->getPath(),
+                    $request->getQueryParams(),
+                    $request->getProtocolVersion(),
+                    $headers
+                );
+            }
+        }
+
+        if ($request->hasHeader('Content-Length')) {
+            $string = $request->getHeaderLine('Content-Length');
+
+            $contentLength = (int)$string;
+            if ((string)$contentLength !== (string)$string) {
+                // Content-Length value is not an integer or not a single integer
+                $this->emit('error', array(new \InvalidArgumentException('The value of `Content-Length` is not valid')));
+                return $this->writeError($conn, 400);
+            }
+
+            $stream = new LengthLimitedStream($conn, $contentLength);
         }
```

Comment:
Not sure I get what you're suggesting here and I may be missing something obvious here, but I'm specifically talking about this:

>    6.  If this is a request message and none of the above are true, then
       the message body length is zero (no message body is present).

https://tools.ietf.org/html/rfc7230#section-3.3.3

Is this correctly covered and tested by this PR or is this not supposed to be part of this PR? See also the PR title and #104.